File: dotnet-elasticemail-examples/Examples/BasicSend.cs
namespace ElasticEmailExamples;

using ElasticEmail.Api;
using ElasticEmail.Client;
using ElasticEmail.Model;

public static class BasicSend
{
    public static async Task RunAsync()
    {
        var emailsApi = new EmailsApi(Ee.Config());

        try
        {
            var result = await emailsApi.EmailsTransactionalPostAsync(new EmailTransactionalMessageData(
                recipients: new TransactionalRecipient(to: new List<string> { Ee.To }),
                content: new EmailContent(
                    from: Ee.From,
                    subject: "Hello from Elastic Email!",
                    body: new List<BodyPart>
                    {
                        new BodyPart(BodyContentType.HTML,
                            "<h1>Welcome!</h1><p>This email was sent using the Elastic Email C# SDK.</p>"),
                        new BodyPart(BodyContentType.PlainText,
                            "Welcome! This email was sent using the Elastic Email C# SDK."),
                    })));

            Console.WriteLine("Email sent successfully!");
            Console.WriteLine($"Transaction ID: {result.TransactionID}");
            Console.WriteLine($"Message ID: {result.MessageID}");
        }
        catch (ApiException e)
        {
            Ee.Fail("send email", e);
        }
    }
}

File: rust-elasticemail-examples/examples/basic_send.rs
use rust_elasticemail_examples::{config, from, print_api_error, to};
use ElasticEmail::apis::emails_api;
use ElasticEmail::models::{BodyContentType, BodyPart, EmailContent, EmailTransactionalMessageData, TransactionalRecipient};

#[tokio::main]
async fn main() {
    let config = config();

    let content = EmailContent {
        subject: Some("Hello from Elastic Email!".to_string()),
        body: Some(vec![
            BodyPart {
                content: Some("<h1>Welcome!</h1><p>This email was sent using the Elastic Email Rust SDK.</p>".to_string()),
                ..BodyPart::new(BodyContentType::Html)
            },
            BodyPart {
                content: Some("Welcome! This email was sent using the Elastic Email Rust SDK.".to_string()),
                ..BodyPart::new(BodyContentType::PlainText)
            },
        ]),
        ..EmailContent::new(from())
    };

    let data = EmailTransactionalMessageData::new(TransactionalRecipient::new(vec![to()]), content);

    match emails_api::emails_transactional_post(&config, data).await {
        Ok(result) => {
            println!("Email sent successfully!");
            println!("Transaction ID: {}", result.transaction_id.unwrap_or_default());
            println!("Message ID: {}", result.message_id.unwrap_or_default());
        }
        Err(e) => {
            print_api_error("sending email", &e);
            std::process::exit(1);
        }
    }
}
